pkg/terra/client/client.go

```diff
@@ -0,0 +1,334 @@
+package client
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	"github.com/smartcontractkit/terra.go/tx"
+	rpchttp "github.com/tendermint/tendermint/rpc/client/http"
+
+	tmtypes "github.com/cosmos/cosmos-sdk/client/grpc/tmservice"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	txtypes "github.com/cosmos/cosmos-sdk/types/tx"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/smartcontractkit/terra.go/msg"
+	wasmtypes "github.com/terra-money/core/x/wasm/types"
+
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	cosmosclient "github.com/cosmos/cosmos-sdk/client"
+	"github.com/pkg/errors"
+	"github.com/terra-money/core/app"
+
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+
+	"github.com/smartcontractkit/terra.go/key"
+)
+
+//go:generate mockery --name ReaderWriter --output ./mocks/
+type ReaderWriter interface {
+	Writer
+	Reader
+}
+
+// Only depends on the cosmos sdk types.
+type Reader interface {
+	GasPrice() sdk.DecCoin
+	Account(address sdk.AccAddress) (uint64, uint64, error)
+	ContractStore(contractAddress string, queryMsg []byte) ([]byte, error)
```

Comment:
One thing that I just noticed is that `contractAddress` here is string and in other method in this same interface it's of type `cosmos-sdk/types#AccAddress`. I'd be nice if this was consistent. CC @connorwstein